app/uk/gov/hmrc/helptosave/util/JsLookupHelper.scala

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmrc.helptosave.util
+
+import play.api.libs.json._
+
+object JsLookupHelper{
+
+  /**
+   * Return the property corresponding to the fieldName, supposing we have a JsObject.
+   *
+   * @param fieldName the name of the property to look up
+   */
+  def lookup(fieldName: String, value: JsValue): JsLookupResult  = JsDefined(value) match {
+    case JsDefined(obj @ JsObject(_)) =>
+      obj.value.get(fieldName).map(JsDefined.apply)
+        .getOrElse(JsUndefined(s"'$fieldName' is undefined on object: ${obj.keys}"))
```

Comment:
It would be nicer to do say: ${obj.keys.map(s"'${_}'").mkString(","}, to be consistent with the rest of the error message string

Reply:
done